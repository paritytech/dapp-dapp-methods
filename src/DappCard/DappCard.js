// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';
import PropTypes from 'prop-types';
import { Accordion, Button, Card, List } from 'semantic-ui-react';
import DappIcon from '@parity/ui/lib/DappIcon';
import methodGroups, { allMethods } from '@parity/mobx/lib/methodGroups';
import { FormattedMessage } from 'react-intl';

import styles from './DappCard.css';

export class DappCard extends Component {
  static propTypes = {
    dapp: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      image: PropTypes.string
    }).isRequired,
    dappsPermissionsStore: PropTypes.object.isRequired,
    editingMode: PropTypes.bool,
    onEdit: PropTypes.func.isRequired,
    onToggle: PropTypes.func.isRequired
  };

  handleEdit = () => this.props.onEdit(this.props.dapp.id);

  handleToggle = (_, { name: method }) => {
    this.props.onToggle(method, this.props.dapp.id);
  };

  renderEditingMode = () => {
    const { dapp, dappsPermissionsStore } = this.props;

    const panels = Object.keys(methodGroups).map(group => ({
      title: {
        key: `${dapp.id}-${group}-editing-title`,
        active: true,
        content: <FormattedMessage {...messages[group]} />
      },
      content: {
        key: `${dapp.id}-${group}-editing-content`,
        active: true,
        content: (
          <List className={styles.editList}>
            {methodGroups[group].methods.map(method => (
              <List.Item
                key={method}
                name={method}
                onClick={this.handleToggle}
                className={styles.editListItem}
              >
                <List.Icon
                  name={
                    dappsPermissionsStore.hasAppPermission(method, dapp.id)
                      ? 'checkmark box'
                      : 'square outline'
                  }
                />
                {method}
              </List.Item>
            ))}
          </List>
        )
      }
    }));

    return <Accordion panels={panels} />;
  };

  renderViewMode = () => {
    const { dappsPermissionsStore, dapp } = this.props;

    if (
      !allMethods.some(method =>
        dappsPermissionsStore.hasAppPermission(method, dapp.id)
      )
    ) {
      return (
        <div className={styles.noAllowedMethods}>
          <FormattedMessage
            id="dapps.methods.noAllowedMethods"
            defaultMessage="No allowed methods"
          />
        </div>
      );
    }

    const panels = Object.keys(methodGroups)
      .filter((
        group // Only display method groups that have at least one method
      ) =>
        methodGroups[group].methods.some(method =>
          dappsPermissionsStore.hasAppPermission(method, dapp.id)
        )
      )
      .map(group => ({
        title: {
          key: `${dapp.id}-${group}-title`,
          content: <FormattedMessage {...messages[group]} />
        },
        content: {
          key: `${dapp.id}-${group}-content`,
          content: (
            <List bulleted className={styles.list}>
              {methodGroups[group].methods.map(
                method =>
                  dappsPermissionsStore.hasAppPermission(method, dapp.id) && (
                    <List.Item key={method}>{method}</List.Item>
                  )
              )}
            </List>
          )
        }
      }));

    return <Accordion panels={panels} />;
  };

  render() {
    const { dapp, editingMode } = this.props;

    return (
      <Card>
        <Card.Content>
          <Button
            floated="right"
            basic
            size="mini"
            icon={editingMode ? 'remove' : 'edit'}
            className={styles.editButton}
            onClick={this.handleEdit}
          />
          <DappIcon
            app={dapp}
            className={`ui image centered ${styles.picture}`}
            size="small"
          />
          <Card.Header>{dapp.name}</Card.Header>
          <Card.Meta>{dapp.description}</Card.Meta>
          <Card.Description>
            {editingMode ? this.renderEditingMode() : this.renderViewMode()}
          </Card.Description>
        </Card.Content>
      </Card>
    );
  }
}

export default inject('dappsPermissionsStore')(observer(DappCard));

const messages = {
  accounts: {
    id: 'dapp.methods.accounts',
    description: 'Explanation of the accounts methodGroup permission',
    defaultMessage: 'Can view your Parity accounts'
  },
  accountsCreate: {
    id: 'dapp.methods.accountsCreate',
    description: 'Explanation of the accountsCreate methodGroup permission',
    defaultMessage: 'Can create new Parity accounts'
  },
  accountsDelete: {
    id: 'dapp.methods.accountsDelete',
    description: 'Explanation of the accountsDelete methodGroup permission',
    defaultMessage: 'Can delete your Parity accounts'
  },
  accountsEdit: {
    id: 'dapp.methods.accountsEdit',
    description: 'Explanation of the accountsEdit methodGroup permission',
    defaultMessage: 'Can edit your Parity accounts'
  },
  dapps: {
    id: 'dapp.methods.dapps',
    description: 'Explanation of the dapps methodGroup permission',
    defaultMessage: 'Can read info about dapps'
  },
  dappsEdit: {
    id: 'dapp.methods.dappsEdit',
    description: 'Explanation of the dappsEdit methodGroup permission',
    defaultMessage: 'Can write info on dapps'
  },
  node: {
    id: 'dapp.methods.node',
    description: 'Explanation of the node methodGroup permission',
    defaultMessage: 'Can read info about node'
  },
  nodeUpgrade: {
    id: 'dapp.methods.nodeUpgrade',
    description: 'Explanation of the nodeUpgrade methodGroup permission',
    defaultMessage: 'Can upgrade Parity'
  },
  shell: {
    id: 'dapp.methods.shell',
    description: 'Explanation of the shell methodGroup permission',
    defaultMessage: 'Can get access to the Shell'
  },
  signerConfirm: {
    id: 'dapp.methods.signerConfirm',
    description: 'Explanation of the signerConfirm methodGroup permission',
    defaultMessage: 'Can confirm or reject requests from the signer'
  },
  signerRequests: {
    id: 'dapp.methods.signerRequests',
    description: 'Explanation of the signerRequests methodGroup permission',
    defaultMessage: 'Can get access to signer requests'
  },
  vaults: {
    id: 'dapp.methods.vaults',
    description: 'Explanation of the vaults methodGroup permission',
    defaultMessage: 'Can get access to your vaults'
  },
  vaultsCreate: {
    id: 'dapp.methods.vaultsCreate',
    description: 'Explanation of the vaultsCreate methodGroup permission',
    defaultMessage: 'Can create new vaults'
  },
  vaultsEdit: {
    id: 'dapp.methods.vaultsEdit',
    description: 'Explanation of the vaultsEdit methodGroup permission',
    defaultMessage: 'Can edit your vaults'
  }
};
