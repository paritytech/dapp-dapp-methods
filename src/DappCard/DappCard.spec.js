// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import PropTypes from 'prop-types';
import { shallowToJson } from 'enzyme-to-json';
import { Button, List } from 'semantic-ui-react';
import DappsPermissionsStore from '@parity/mobx/lib/dapps/DappsPermissionsStore';

import { shallowWithIntl, mountWithIntl } from '../setupTests';
import { DappCard } from './DappCard';

const mockPermissions = { 'shell_loadApp:123': true };
const mockApi = {
  shell: { getMethodPermissions: () => Promise.resolve(mockPermissions) }
};
const props = {
  dapp: {
    id: '123',
    name: 'Test'
  },
  dappsPermissionsStore: DappsPermissionsStore.get(mockApi),
  onEdit: () => {},
  onToggle: () => {}
};

test('should render correctly in viewing mode', () => {
  const component = shallowWithIntl(<DappCard {...props} />);

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should render correctly in viewing mode with no allowed methods', () => {
  const dappsPermissionsStore = new DappsPermissionsStore({
    shell: {
      ...mockApi.shell,
      getMethodPermissions: () => Promise.resolve({}) // Return {} from api
    }
  });
  const component = shallowWithIntl(
    <DappCard {...props} dappsPermissionsStore={dappsPermissionsStore} />
  );

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should render correctly in editing mode', () => {
  const component = shallowWithIntl(<DappCard {...props} editingMode />);

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should handle onEdit button click', () => {
  const onEdit = jest.fn();

  const component = shallowWithIntl(<DappCard {...props} onEdit={onEdit} />);

  component
    .find(Button)
    .last()
    .simulate('click');
  expect(onEdit).toHaveBeenCalledWith('123');
});

test('should handle onToggle click', () => {
  const onToggle = jest.fn();

  const component = mountWithIntl(
    <DappCard {...props} editingMode onToggle={onToggle} />,
    {
      context: { api: mockApi },
      childContextTypes: { api: PropTypes.object.isRequired }
    }
  );

  component
    .find(List.Item)
    .first() // The 1st one in methodGroup.js is 'shell_loadApp'
    .simulate('click');

  expect(onToggle).toHaveBeenCalledWith('shell_loadApp', '123');
});
