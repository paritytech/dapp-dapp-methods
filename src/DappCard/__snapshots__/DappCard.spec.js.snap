// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly in editing mode 1`] = `
<Card>
  <CardContent>
    <Button
      as="button"
      basic={true}
      floated="right"
      icon="remove"
      onClick={[Function]}
      size="mini"
    />
    <DappIcon
      app={
        Object {
          "id": "123",
          "name": "Test",
        }
      }
      className="ui image centered undefined"
      raised={true}
      size="small"
    />
    <CardHeader>
      Test
    </CardHeader>
    <CardMeta />
    <CardDescription>
      <Accordion
        panels={
          Array [
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="shell_loadApp"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="checkmark box"
                          />
                          shell_loadApp
                  </ListItem>
          </List>,
                "key": "123-shell-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can get access to the Shell"
                  description="Explanation of the shell methodGroup permission"
                  id="dapp.methods.shell"
                  values={Object {}}
          />,
                "key": "123-shell-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_dappsRefresh"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_dappsRefresh
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_dappsUrl"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_dappsUrl
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="shell_getApps"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          shell_getApps
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="shell_getMethodPermissions"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          shell_getMethodPermissions
                  </ListItem>
          </List>,
                "key": "123-dapps-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can read info about dapps"
                  description="Explanation of the dapps methodGroup permission"
                  id="dapp.methods.dapps"
                  values={Object {}}
          />,
                "key": "123-dapps-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="shell_setAppPinned"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          shell_setAppPinned
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="shell_setAppVisibility"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          shell_setAppVisibility
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="shell_setMethodPermissions"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          shell_setMethodPermissions
                  </ListItem>
          </List>,
                "key": "123-dappsEdit-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can write info on dapps"
                  description="Explanation of the dappsEdit methodGroup permission"
                  id="dapp.methods.dappsEdit"
                  values={Object {}}
          />,
                "key": "123-dappsEdit-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_accountsInfo"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_accountsInfo
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_allAccountsInfo"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_allAccountsInfo
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_getNewDappsAddresses"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_getNewDappsAddresses
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_getNewDappsDefaultAddress"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_getNewDappsDefaultAddress
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_hardwareAccountsInfo"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_hardwareAccountsInfo
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_lockedHardwareAccountsInfo"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_lockedHardwareAccountsInfo
                  </ListItem>
          </List>,
                "key": "123-accounts-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can view your Parity accounts"
                  description="Explanation of the accounts methodGroup permission"
                  id="dapp.methods.accounts"
                  values={Object {}}
          />,
                "key": "123-accounts-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_generateSecretPhrase"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_generateSecretPhrase
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_importGethAccounts"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_importGethAccounts
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_listGethAccounts"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_listGethAccounts
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_newAccountFromPhrase"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_newAccountFromPhrase
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_newAccountFromSecret"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_newAccountFromSecret
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_newAccountFromWallet"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_newAccountFromWallet
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_phraseToAddress"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_phraseToAddress
                  </ListItem>
          </List>,
                "key": "123-accountsCreate-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can create new Parity accounts"
                  description="Explanation of the accountsCreate methodGroup permission"
                  id="dapp.methods.accountsCreate"
                  values={Object {}}
          />,
                "key": "123-accountsCreate-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_setAccountName"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_setAccountName
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_setAccountMeta"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_setAccountMeta
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_hardwarePinMatrixAck"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_hardwarePinMatrixAck
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_setNewDappsAddresses"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_setNewDappsAddresses
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_setNewDappsDefaultAddress"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_setNewDappsDefaultAddress
                  </ListItem>
          </List>,
                "key": "123-accountsEdit-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can edit your Parity accounts"
                  description="Explanation of the accountsEdit methodGroup permission"
                  id="dapp.methods.accountsEdit"
                  values={Object {}}
          />,
                "key": "123-accountsEdit-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_killAccount"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_killAccount
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_removeAddress"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_removeAddress
                  </ListItem>
          </List>,
                "key": "123-accountsDelete-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can delete your Parity accounts"
                  description="Explanation of the accountsDelete methodGroup permission"
                  id="dapp.methods.accountsDelete"
                  values={Object {}}
          />,
                "key": "123-accountsDelete-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_closeVault"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_closeVault
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_getVaultMeta"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_getVaultMeta
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_listVaults"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_listVaults
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_listOpenedVaults"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_listOpenedVaults
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_openVault"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_openVault
                  </ListItem>
          </List>,
                "key": "123-vaults-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can get access to your vaults"
                  description="Explanation of the vaults methodGroup permission"
                  id="dapp.methods.vaults"
                  values={Object {}}
          />,
                "key": "123-vaults-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_newVault"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_newVault
                  </ListItem>
          </List>,
                "key": "123-vaultsCreate-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can create new vaults"
                  description="Explanation of the vaultsCreate methodGroup permission"
                  id="dapp.methods.vaultsCreate"
                  values={Object {}}
          />,
                "key": "123-vaultsCreate-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_changeVault"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_changeVault
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_changeVaultPassword"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_changeVaultPassword
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_setVaultMeta"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_setVaultMeta
                  </ListItem>
          </List>,
                "key": "123-vaultsEdit-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can edit your vaults"
                  description="Explanation of the vaultsEdit methodGroup permission"
                  id="dapp.methods.vaultsEdit"
                  values={Object {}}
          />,
                "key": "123-vaultsEdit-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_checkRequest"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_checkRequest
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_localTransactions"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_localTransactions
                  </ListItem>
          </List>,
                "key": "123-signerRequests-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can get access to signer requests"
                  description="Explanation of the signerRequests methodGroup permission"
                  id="dapp.methods.signerRequests"
                  values={Object {}}
          />,
                "key": "123-signerRequests-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_confirmRequest"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_confirmRequest
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_confirmRequestRaw"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_confirmRequestRaw
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_rejectRequest"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_rejectRequest
                  </ListItem>
          </List>,
                "key": "123-signerConfirm-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can confirm or reject requests from the signer"
                  description="Explanation of the signerConfirm methodGroup permission"
                  id="dapp.methods.signerConfirm"
                  values={Object {}}
          />,
                "key": "123-signerConfirm-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_hashContent"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_hashContent
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_consensusCapability"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_consensusCapability
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_upgradeReady"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_upgradeReady
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_versionInfo"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_versionInfo
                  </ListItem>
                  <ListItem
                          className={undefined}
                          name="parity_wsUrl"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_wsUrl
                  </ListItem>
          </List>,
                "key": "123-node-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can read info about node"
                  description="Explanation of the node methodGroup permission"
                  id="dapp.methods.node"
                  values={Object {}}
          />,
                "key": "123-node-editing-title",
              },
            },
            Object {
              "content": Object {
                "active": true,
                "content": <List
                  className={undefined}
          >
                  <ListItem
                          className={undefined}
                          name="parity_executeUpgrade"
                          onClick={[Function]}
                  >
                          <ListIcon
                                  name="square outline"
                          />
                          parity_executeUpgrade
                  </ListItem>
          </List>,
                "key": "123-nodeUpgrade-editing-content",
              },
              "title": Object {
                "active": true,
                "content": <FormattedMessage
                  defaultMessage="Can upgrade Parity"
                  description="Explanation of the nodeUpgrade methodGroup permission"
                  id="dapp.methods.nodeUpgrade"
                  values={Object {}}
          />,
                "key": "123-nodeUpgrade-editing-title",
              },
            },
          ]
        }
      />
    </CardDescription>
  </CardContent>
</Card>
`;

exports[`should render correctly in viewing mode 1`] = `
<Card>
  <CardContent>
    <Button
      as="button"
      basic={true}
      floated="right"
      icon="edit"
      onClick={[Function]}
      size="mini"
    />
    <DappIcon
      app={
        Object {
          "id": "123",
          "name": "Test",
        }
      }
      className="ui image centered undefined"
      raised={true}
      size="small"
    />
    <CardHeader>
      Test
    </CardHeader>
    <CardMeta />
    <CardDescription>
      <Accordion
        panels={
          Array [
            Object {
              "content": Object {
                "content": <List
                  bulleted={true}
                  className={undefined}
          >
                  <ListItem>
                          shell_loadApp
                  </ListItem>
          </List>,
                "key": "123-shell-content",
              },
              "title": Object {
                "content": <FormattedMessage
                  defaultMessage="Can get access to the Shell"
                  description="Explanation of the shell methodGroup permission"
                  id="dapp.methods.shell"
                  values={Object {}}
          />,
                "key": "123-shell-title",
              },
            },
          ]
        }
      />
    </CardDescription>
  </CardContent>
</Card>
`;

exports[`should render correctly in viewing mode with no allowed methods 1`] = `
<Card>
  <CardContent>
    <Button
      as="button"
      basic={true}
      floated="right"
      icon="edit"
      onClick={[Function]}
      size="mini"
    />
    <DappIcon
      app={
        Object {
          "id": "123",
          "name": "Test",
        }
      }
      className="ui image centered undefined"
      raised={true}
      size="small"
    />
    <CardHeader>
      Test
    </CardHeader>
    <CardMeta />
    <CardDescription>
      <div>
        <FormattedMessage
          defaultMessage="No allowed methods"
          id="dapps.methods.noAllowedMethods"
          values={Object {}}
        />
      </div>
    </CardDescription>
  </CardContent>
</Card>
`;
